SmartDeviceLinkTests/DevAPISpecs/SDLChoiceSetManagerSpec.m

```diff
@@ -440,105 +457,384 @@ - (void)sdl_addUniqueNamesToCells:(NSMutableSet<SDLChoiceCell *> *)choices;
+                    return [value isKindOfClass:[SDLPreloadChoicesOperation class]];
+                }]]);
+                OCMExpect([strickMockOperationQueue addOperation:[OCMArg checkWithBlock:^BOOL(id value) {
+                    SDLPresentChoiceSetOperation *presentChoicesOperation = (SDLPresentChoiceSetOperation *)value;
+                    presentChoicesOperation.selectedCell = testSelectedCell;
+                    presentChoicesOperation.selectedTriggerSource = testMode;
+                    presentChoicesOperation.selectedCellRow = testSelectedCellRow;
+                    presentChoicesOperation.internalError = nil;
+                    presentChoicesOperation.completionBlock();
+                    return [value isKindOfClass:[SDLPresentChoiceSetOperation class]];
+                }]]);
+                OCMExpect([choiceDelegate choiceSet:testChoiceSet didSelectChoice:testSelectedCell withSource:testMode atRowIndex:testSelectedCellRow]);
+
+                [testManager presentChoiceSet:testChoiceSet mode:testMode withKeyboardDelegate:keyboardDelegate];
+
+                OCMVerifyAllWithDelay(strickMockOperationQueue, 0.5);
+                OCMVerifyAllWithDelay(choiceDelegate, 0.5);
+
+                expect(testManager.pendingPresentationSet).to(beNil());
+                expect(testManager.pendingPresentOperation).to(beNil());
+
+                expect(testManager.preloadedMutableChoices.count).to(equal(3));
+                expect(testManager.pendingMutablePreloadChoices).to(beEmpty());
+
+                // Present the exact same choices again
+                OCMReject([strickMockOperationQueue addOperation:[OCMArg isKindOfClass:SDLPreloadChoicesOperation.class]]);
+                OCMExpect([strickMockOperationQueue addOperation:[OCMArg checkWithBlock:^BOOL(id value) {
+                    SDLPresentChoiceSetOperation *presentChoicesOperation = (SDLPresentChoiceSetOperation *)value;
+                    presentChoicesOperation.selectedCell = testSelectedCell;
+                    presentChoicesOperation.selectedTriggerSource = testMode;
+                    presentChoicesOperation.selectedCellRow = testSelectedCellRow;
+                    presentChoicesOperation.internalError = nil;
+                    presentChoicesOperation.completionBlock();
+                    return [value isKindOfClass:[SDLPresentChoiceSetOperation class]];
+                }]]);
+                OCMExpect([choiceDelegate choiceSet:testChoiceSet didSelectChoice:testSelectedCell withSource:testMode atRowIndex:testSelectedCellRow]);
+
+                [testManager presentChoiceSet:testChoiceSet mode:testMode withKeyboardDelegate:keyboardDelegate];
+
+                OCMVerifyAllWithDelay(strickMockOperationQueue, 0.5);
+                OCMVerifyAllWithDelay(choiceDelegate, 0.5);
+            });
+
+            it(@"It should upload choices that failed to upload in previous presentations", ^{
```

Comment:
```suggestion
            it(@"should upload choices that failed to upload in previous presentations", ^{
```

I think this needs more information. It shouldn't upload them unless another presentation is created, right?